compiler/rustc_smir/src/rustc_internal/pretty.rs

```diff
@@ -0,0 +1,291 @@
+use std::io;
+
+use rustc_middle::ty::TyCtxt;
+use stable_mir::{
+    mir::{Mutability, Operand, Rvalue, StatementKind},
+    ty::{RigidTy, TyKind},
+    CrateItem,
+};
+
+use super::{internal, run};
+
+pub fn write_smir_pretty<'tcx>(tcx: TyCtxt<'tcx>, w: &mut dyn io::Write) -> io::Result<()> {
+    writeln!(w, "// WARNING: This is highly experimental output it's intended for stable-mir developers only.").unwrap();
+    writeln!(w, "// If you find a bug or want to improve the output open a issue at https://github.com/rust-lang/project-stable-mir.").unwrap();
+    run(tcx, || {
+        let items = stable_mir::all_local_items();
+        items.iter().for_each(|item| {
+            // Because we can't return a Result from a closure, we have to unwrap here.
+            writeln!(w, "{}", function_name(*item, tcx)).unwrap();
+            writeln!(w, "{}", function_body(*item, tcx)).unwrap();
+            item.body().blocks.iter().enumerate().for_each(|(index, block)| {
+                writeln!(w, "    bb{}: {{", index).unwrap();
+                block.statements.iter().for_each(|statement| {
+                    writeln!(w, "{}", pretty_statement(&statement.kind, tcx)).unwrap();
+                });
+                writeln!(w, "    }}").unwrap();
```

Comment:
I think it would be super handy to have a function in `stable_mir` that emits a function item that users can use to dump the stable-mir they are analyzing. Do you mind adding one?

Reply:
I don't really understand this can you provide pseudo code of what this function going to do ?

Reply:
I added the suggestion below.